cpp/mg_utility/mage.hpp

```diff
@@ -0,0 +1,2757 @@
+  /// only the last field may be a non-integer.
+  explicit Duration(std::string_view string) : ptr_(mgp::duration_from_string(string.data(), memory)) {}
+
+  /// @brief Creates a Duration object from the given number of microseconds.
+  explicit Duration(int64_t microseconds) : ptr_(mgp::duration_from_microseconds(microseconds, memory)) {}
+
+  /// @brief Creates a Duration object with the given `day`, `hour`, `minute`, `second`, `millisecond`, and
+  /// `microsecond` properties.
+  Duration(double day, double hour, double minute, double second, double millisecond, double microsecond);
+
+  Duration(const Duration &other) : Duration(other.ptr_) {}
+  Duration(Duration &&other) : ptr_(other.ptr_) { other.ptr_ = nullptr; };
+
+  Duration &operator=(const Duration &other) = delete;
+  Duration &operator=(Duration &&other) = delete;
+
+  ~Duration();
+
+  /// @brief Returns the duration as microseconds.
+  int64_t microseconds() const { return mgp::duration_get_microseconds(ptr_); }
+
+  bool operator==(const Duration &other) const;
+  Duration operator+(const Duration &other) const;
+  Duration operator-(const Duration &other) const;
+  Duration operator-() const;
+
+  bool operator<(const Duration &other) const;
+
+ private:
+  mgp_duration *ptr_;
+};
+/* #endregion */
+
+/* #endregion */
+
+/* #region Value */
+enum class Type : uint8_t {
+  Null,
+  Bool,
+  Int,
+  Double,
+  String,
+  List,
+  Map,
+  Node,
+  Relationship,
+  Path,
+  Date,
+  LocalTime,
+  LocalDateTime,
+  Duration,
+  UNDEF,  // for indicating an undefined value
+};
+
+/// @brief Wrapper class for @ref mgp_value.
+class Value {
+ public:
+  friend class List;
+  friend class Map;
+  friend class Date;
+  friend class LocalTime;
+  friend class LocalDateTime;
+  friend class Duration;
+  friend class Record;
+  friend class Result;
+
+  explicit Value(mgp_value *ptr) : ptr_(mgp::value_copy(ptr, memory)) {}
+
+  // Null constructor:
+  explicit Value() : ptr_(mgp::value_make_null(memory)) {}
+
+  // Primitive type constructors:
+  explicit Value(const bool value) : ptr_(mgp::value_make_bool(value, memory)) {}
+  explicit Value(const int64_t value) : ptr_(mgp::value_make_int(value, memory)) {}
+  explicit Value(const double value) : ptr_(mgp::value_make_double(value, memory)) {}
+
+  // String constructors:
+  explicit Value(const char *value) : ptr_(mgp::value_make_string(value, memory)) {}
+  explicit Value(const std::string_view value) : ptr_(mgp::value_make_string(value.data(), memory)) {}
+
+  // Container constructors:
+
+  /// @brief Constructs a List value from the copy of the given `list`.
+  explicit Value(const List &list) : ptr_(mgp::value_make_list(list.ptr_)) {}
+  /// @note The behavior of accessing `list` after performing this operation is undefined.
+  explicit Value(List &&list) {
+    ptr_ = mgp::value_make_list(list.ptr_);
+    delete &list;
```

Comment:
This looks scary 😂 Haven't seen this pattern before? @antaljanosbenjamin can maybe leave an opinion/guide here?

Reply:
This part is inherited from [Matak’s API concept](#68); it looked fine to me, but Benjo’s opinion is always appreciated! :slightly_smiling_face: